### PR TITLE
US UK AU decorators

### PR DESCRIPTION
**Pull request by ChrisH91**

### CODE REVIEW

File: src/decorators/gb.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      localNumber = `${phoneNumber.substring(3, 5)}` +
+        `${phoneNumber.length > 3 ? ` ${phoneNumber.substring(5, 9)}` : ""}` +
+        `${phoneNumber.length > 5 ? ` ${phoneNumber.substring(9)}` : ""}`;
+    } else if (
+      phoneNumber.substring(2, 4) === "03" ||
+      phoneNumber.substring(2, 4) === "08" ||
+      phoneNumber.substring(2, 4) === "09" ||
+      phoneNumber.substring(2, 4) === "08" ||
+      phoneNumber.substring(2, 5) === "011" ||
+      (phoneNumber.substring(2, 4) === "01" && phoneNumber.charAt(5) === "1")
```
Comment (brgmnn) [inline src/decorators/gb.ts:33]: Are these files hand written? It seems a bit ad hoc to write these long conditional blocks for specific countries. It's good as a starting point though.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import IDecoratedNumber from "./decorators/decorated-number";
+import decorators from "./decorators/index";
+import detectCountry from "./detect-country";
+
+export default {
+  decorate: (phoneNumber: string): IDecoratedNumber => {
```
Comment (brgmnn) [inline src/index.ts:6]: It's assumed that this is an unformatted number? For instance `61023333444` (from the tests). Would we want to do any basic whitespace stripping or removing a trailing `+`?

File: src/data/country-code-to-dialing-code.json [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+  "AT": "43",
+  "AW": "297",
+  "IN": "91",
+  "AZ": "994",
+  "IE": "353",
+  "ID": "62",
+  "UA": "380",
+  "QA": "974",
+  "MZ": "258"
+}
```
Comment (JPeaker) [inline src/data/country-code-to-dialing-code.json:252]: You have a file going one way, then another file going the other. If you had to add any countries, you'd have to update them in both places. Can't we just define it in one place, and have the other object just defined by a kind of "object reversal", swapping keys and values?

File: src/decorators/gb.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  "017684",
+  "017687",
+  "019467",
+];
+
+const decorator: IDecorator = {
+  decorate(phoneNumber: string) {
+    let localNumber;
+
+    if (phoneNumber.substring(2, 4) === "02") {
```
Comment (JPeaker) [inline src/decorators/gb.ts:23]: Sounds to me like it'd be useful to chuck `phoneNumber.substring(2, 4)` in a constant, given how many times you're checking it in this function. nbd though

**Review by brgmnn — APPROVED** [review]
What's the goals regarding things like as-you-type, extensions and e163/e164 format?

**Review by JPeaker — CHANGES_REQUESTED** [review]

File: src/data/country-code-to-dialing-code.json (hunk 3 above)
Comment (ChrisH91) [reply src/data/country-code-to-dialing-code.json:252]: It's not as simple as that unfortunately.

Each country code will have exactly 1 dialing code, but 1 dialing code may represent multiple countries.

File: src/decorators/gb.ts (hunk 1 above)
Comment (ChrisH91) [reply src/decorators/gb.ts:33]: I'm not sure there will be another way around it.

Each country has different validation and formatting rules for their phone numbers, And a lot of the rules are antiquated or without much logical sense to them (though I have a feeling that the UK one will be the worst).

File: src/index.ts (hunk 2 above)
Comment (ChrisH91) [reply src/index.ts:6]: Yeah, we will strip out any non-numeric characters before we call decorate. I'm saving that for a future PR though.

There's a lot of logic around the way people specify international dialing, as well as local leading digits and what not.

**Comment by ChrisH91** [comment]
@brgmnn E164 will be coming in a later PR.

**Review by JPeaker — APPROVED** [review]

**Pull request merged**